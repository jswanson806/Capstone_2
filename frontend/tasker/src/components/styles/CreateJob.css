.createJob-container {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 0 2em;
  }
  
  .createJob-centered-card {
    max-height: 700px;
    width: 100%;
    margin: 0 auto;
    background-color: #fff;
    border-radius: 20px;
    padding: 20px;
    transition: max-height 0.2s ease-in-out;
  }
  
  .createJob-centered-card h5 {
    text-align: center;
    margin-bottom: 20px; /* Add spacing below the title */
  }
  
  .createJob-centered-card input {
    width: 100%;
    padding: 2%;
    margin-bottom: 10px;
    border-radius: 5px;
  }
  
  .createJob-centered-card .button {
    margin: 2px;
    width: 100%;
    padding: 10px;
    color: #fff; /* Button text color */
    border: none;
    border-radius: 5px;
    cursor: pointer;
  
  }
  
  .modal-content {
    background-color: transparent !important;
    border: none !important;
  }