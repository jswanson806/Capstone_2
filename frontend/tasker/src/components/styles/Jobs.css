.jobs-header-container {
    text-align: center;
    margin: 2% 2%;
}

.jobs-buttons-container {
    padding-bottom: 2rem;
    text-align: center;
}

.jobs-button {
    margin: 2px 5px;
    padding: 5px;
}

.jobs-jobCard-container {
    justify-content: center;
    padding: 20px;
}

.jobs-notFound-message {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 10vh;
}

.jobs-dynamic-page-header {
    font-weight: bold;
    
}

.jobCard-card {
    height: 15em;
}

.jobCard-badge {
    position: static !important;
    padding: 5px !important;
}

.jobCard-badge-container {
  position: relative;
  top: 34%;
}

p {
    position: relative;
    top: 25%;
}

h5 {
    position: relative;
    top: 25%;
}

.item {
    flex: 1 0 20%;
    max-width: 20%;
}