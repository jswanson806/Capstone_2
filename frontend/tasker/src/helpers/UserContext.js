import React, { createContext, useState } from 'react';

// Create the context
const UserContext = createContext();

// Create a provider for the context
const UserProvider = ({ children }) => {
  const [user, setUser] = useState(localStorage.getItem('user') || '');

  // Function to update the user in the state and localStorage
  const updateUser = (newUser) => {
    setUser(newUser);
    localStorage.setItem('user', newUser);
  };

  return (
    <UserContext.Provider value={{ user, updateUser }}>
      {children}
    </UserContext.Provider>
  );
};

export { UserContext, UserProvider };
