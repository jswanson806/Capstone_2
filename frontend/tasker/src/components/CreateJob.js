import React, { useState, useContext, useRef } from "react";
import { UserContext } from "../helpers/UserContext";
import TaskerApi from "../api";
import { Button, Input, Form, FormGroup, Label, FormText, FormFeedback, Modal, ModalBody, ModalHeader} from "reactstrap";
import "./styles/CreateJob.css";

const CreateJob = ({onCreate, onClose, isOpen}) => {

    const INITIAL_STATE = {
        title: '',
        body: '',
        address: '',
        postedBy: '',
        before_image_url: '',
    }

    const [formData, setFormData] = useState(INITIAL_STATE);
    const [isValid, setIsValid] = useState(true);
    const [imageFile, setImageFile] = useState('');

    const { user } = useContext(UserContext);

    const validFileTypes = ['image/jpg', 'image/jpeg', 'image/png'];

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))

        if(name === "body") {
            adjustTextareaHeight(e.target);
            adjustAreaHeight();
        }
    };

    const adjustTextareaHeight = (textarea) => {
        textarea.style.height = 'auto'; // Reset the height to auto
        textarea.style.height = textarea.scrollHeight + 'px'; // Set the height to the scrollHeight
      };

    const adjustAreaHeight = () => {

        const cardContent = document.querySelector('.createJob-centered-card');
        const textarea = document.querySelector('#body');  

        if (cardContent && textarea) {
          cardContent.style.maxHeight = `${textarea.scrollHeight + 600}px`;
      }
    };

    const handleSubmit = async (e) => {
        e.preventDefault();

        try {
            const jobInfo = {
                job: {
                    title: formData.title,
                    body: formData.body,
                    address: formData.address,
                    posted_by: JSON.parse(user).id,
                    before_image_url: imageFile.name,
                }
            };
            
            const form = new FormData();
            form.append('image', imageFile);

            await TaskerApi.createJob(jobInfo);
            await TaskerApi.uploadBeforeImage(form, JSON.parse(user).id,);

            setFormData(INITIAL_STATE);
            onCreate();
        } catch (error) {
            console.error('Error:', error);
        }
    };
    
    const handleUpload = async (e) => {
    
        const file = e.target.files[0];

        if(!validFileTypes.find(type => type === file.type)){
            setIsValid(false);
            return;
        };

        setIsValid(true);

        setImageFile(file);

    };

    return (
      <div className={`custom-modal ${isOpen ? 'show' : ''}`}>
        <div className="modal-content">
          
          <h2>Create a New Job</h2>

            <Form onSubmit={handleSubmit}>
              <FormGroup>
                  <Label for="title">Job Title:</Label>
                  <Input
                    id="title"
                    type="text"
                    name="title"
                    placeholder="Job Title"
                    data-testid="createJob-form-title-input"
                    value={FormData.title}
                    onChange={handleChange}
                  />
              </FormGroup>
              <FormGroup>
                  <Label for="body">Job Description:</Label>
                  <Input
                    type="textarea"
                    id="body"
                    name="body"
                    placeholder="Briefly describe the job..."
                    data-testid="createJob-form-body-input"
                    value={FormData.body}
                    onChange={handleChange}
                    rows={1}
                    style={{resize: 'none', overflowY: 'hidden'}}
                  />
              </FormGroup>
              <FormGroup>
                  <Label for="address">Address:</Label>
                  <Input
                    id="address"
                    type="text"
                    name="address"
                    placeholder="Address"
                    data-testid="createJob-form-address-input"
                    value={FormData.address}
                    onChange={handleChange}
                  />
                <FormFeedback>Please enter an address.</FormFeedback>
              </FormGroup>
              <FormGroup>
                  <Label for="imageInput">
                      Upload "Before" Image
                  </Label>
                  <Input
                      id="imageInput"
                      type="file"
                      name="image"
                      onChange={handleUpload}
                      valid={isValid}
                      invalid={!isValid}
                  />
                  {!isValid 
                      ? 
                      <FormFeedback >Invalid file type</FormFeedback> 
                      : 
                      <FormFeedback valid>Valid file type</FormFeedback>
                  }
                  <FormText>Supported File Types: jpg, png</FormText>
              </FormGroup>
              <div className="button-container">
              <Button className="button" color="info" type="submit" data-testid="createJob-submit-button">Post Job</Button>
              <Button className="button" color="danger" type="submit" data-testid="createJob-close-button" onClick={onClose}>Close</Button>
              </div>
            </Form>
          </div>
        </div>
    );
  }

export default CreateJob;