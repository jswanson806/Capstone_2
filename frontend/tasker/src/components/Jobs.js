import React, { useContext, useEffect, useState } from 'react';
import TaskerApi from '../api';
import JobCard from './JobCard.js';
import { UserContext } from '../helpers/UserContext';
import CreateJob from './CreateJob.js';
import {
    Spinner, 
    Button, 
    Modal,
    ModalHeader,
    ModalBody,
    Row,
    Container
} from "reactstrap";
import "./styles/Jobs.css";

const Jobs = () => {

    const jobCardsInitialState = [];
    const jobsInitialState = '';

    const {user} = useContext(UserContext)

    const [jobs, setJobs] = useState(jobsInitialState);
    const [jobCards, setJobCards] = useState(jobCardsInitialState);
    const [currUser, setCurrUser] = useState(null);
    const [header, setHeader] = useState('');
    const [buttons, setButtons] = useState('');
    const [isLoading, setIsLoading] = useState(true);
    const [triggerEffect, setTriggerEffect] = useState(false);
    const [notFound, setNotFound] = useState(null);

    // view toggles
    const [isWorker, setIsWorker] = useState(false);
    const [showCreateJob, setShowCreateJob] = useState(false);
    
    /** On initial render
     * 
     *  Calls functions to set state of currUser
     *  Gets the current user from the database
     */
    useEffect(() => {
        // parsed user string received from user context
        const parsedUser = JSON.parse(user);
        // call api to get user from db and set state of currUser
        getCurrUser(parsedUser.id);
        
    }, []);

    
    /** Sets the default content for the page depending on isWorker boolean from the currUser state
     * 
     * Calls API for initial list of jobs
     */
    useEffect(() => {
        
        // if currUser is populated & create job form is not shown
        if(currUser && !showCreateJob){

            if(!isWorker){ // if user is not worker, get the jobs posted by this user
                getAndSetActiveUserJobs(currUser.id);
                // set buttons for users
                setButtons(renderUserButtons());
                // set header for page
                renderHeader('My Active Jobs');
            } else { // else the user is a worker, get all available jobs
                getAndSetAllAvailableJobs();
                // set buttons for workers
                setButtons(renderWorkerButtons());
                // set header for page
                renderHeader('Available Jobs');
            }
        }

    }, [
        currUser,
        isWorker,
        triggerEffect
    ]);


    /** Calls functions to create and set job cards, state of notFound, and set state of isLoading
     *  
     * Waits for jobs and currUser to be populated
     * 
    */
    useEffect(() => {

        // only call once jobs and currUser have been populated
        if(jobs !== jobsInitialState && currUser) {
            // create the job cards
            createJobCards();
            // update state of notFound
            setNotFound(false);
        } else { // jobCards state is default
            // update state of notFound
            setNotFound(true);
        }
        
    }, [ jobs, currUser ]);

    /** Updates state of isLoading once state of notFound is updated */
    useEffect(() => {
        // notFound can either be truthy or falsy to trigger isLoading state update
        if(notFound || notFound === false){
            // set isLoading false
            setIsLoading(false);
        }
    }, [notFound])


    /**
    * Fetches data for a single user from the Tasker API based on the given ID.
    *
    * Updates state of currUser with the response object
    * Updates isWorker with the isWorker property of the response object
    */
    async function getCurrUser(id){
        try {
            const res = await TaskerApi.getSingleUser(id);
            const { user } = res;
            setCurrUser(user);
            setIsWorker(user.isWorker);
        
        } catch(err) {
            console.error(err);
        }
    }

    /**
    * Sets state of JobCards based on current view toggles.
    * 
    * Uses filtering based on the current user, view toggle states, and existing filtered jobs to map JobCard components to jobCards state
    * 
    */
    async function createJobCards(){
        // convert user applications to Set
        const applications = new Set(currUser.applications);
        // map each job to a JobCard componenet
        const jobCards = jobs.map((job) => {
            return (
                
                    <JobCard
                        user={currUser}
                        applications={applications}
                        job={job}
                        getCurrUser={getCurrUser}
                        setJobs={setJobs}
                        key={job.id}
                        triggerEffect={() => setTriggerEffect(!triggerEffect)}
                        data-testid="jobCard-component"
                    />
                
            );
        });

        setJobCards(jobCards);
    };

    /** 
     * Fetches data for jobs posted by the user where the job status is 'active'
     * 
     * Calls api to findInProgressUserJobs
     * 
     * Updates state of jobs with the response array of jobs
     */
    async function getAndSetActiveUserJobs(userId){
        setIsLoading(true);
        try {
            const inProgressUserJobs = await TaskerApi.getActiveUserJobs(userId);
            if(!inProgressUserJobs.jobs) {
                setJobs(jobsInitialState);
                setJobCards(jobCardsInitialState);
            } else {
                const { jobs } = inProgressUserJobs;
                setJobs(jobs);
            };
        } catch(err) {
            console.error(err);
        }
    }

    /**
     * Fetches data for jobs posted by the user where the job status is 'pending review'
     * 
     * Calls api to findPendingReviewUserJobs
     * 
     * Updates state of jobs with the response array of jobs
    */
    async function getAndSetPendingReviewUserJobs(userId){
        setIsLoading(true);
        try {
            const allPendingReviewUserJobs = await TaskerApi.getPendingReviewUserJobs(userId);
            if(!allPendingReviewUserJobs.pendingReviewUserJobs) {
                setJobs(jobsInitialState);
                setJobCards(jobCardsInitialState);
            } else {   
                const { pendingReviewUserJobs } = allPendingReviewUserJobs;
                setJobs(pendingReviewUserJobs);
            }
        } catch(err) {
            console.error(err);
        }
    }

    /**
     * Fetches data for jobs available for applications by workers
     * 
     * Calls api to findAllAvailableJobs
     * 
     * Updates state of jobs with the response array of jobs
    */
    async function getAndSetAllAvailableJobs(){
        setIsLoading(true);
        try {
            const allAvailableJobs = await TaskerApi.getAllAvailableJobs();
            const { allJobs } = allAvailableJobs;
            setJobs(allJobs);
        } catch(err) {
            console.error(err);
        }
    }

    /**
     * Fetches data for jobs to which worker has already applied
     * 
     * Calls api to findAllAppliedWorkerJobs
     * 
     * Updates state of jobs with the response array of jobs
    */
    async function getAndSetAllAppliedWorkerJobs(workerId){
        setIsLoading(true);
        try {
            const allAppliedJobs = await TaskerApi.getAppliedWorkerJobs(workerId);
            if(!allAppliedJobs.appliedJobs){
                setJobs(jobsInitialState);
                setJobCards(jobCardsInitialState);
            } else {
                const { appliedJobs } = allAppliedJobs;
                setJobs(appliedJobs);
            }
        } catch(err) {
            console.error(err);
        }
    }

    /**
     * Fetches data for jobs to which worker has already been assigned
     * 
     * Calls api to findAllInProgressWorkerJobs
     * 
     * Updates state of jobs with the response array of jobs
    */
    async function getAndSetAllAssignedWorkerJobs(workerId){
        setIsLoading(true);
        try {
            const allAssignedWorkerJobs = await TaskerApi.getAllAssignedWorkerJobs(workerId);
            if(!allAssignedWorkerJobs.assignedJobs){
                setJobs(jobsInitialState);
                setJobCards(jobCardsInitialState);
            } else {
                const { assignedJobs } = allAssignedWorkerJobs;
                setJobs(assignedJobs);
            }
        } catch(err) {
            console.error(err);
        }
    }
 
    const toggleCreateJob = () => {
        setShowCreateJob(!showCreateJob);
    }

    const renderHeader = (pageHeader) => {
        setHeader(<h2 className="jobs-dynamic-page-header">{pageHeader}</h2>)
    }


    const renderWorkerButtons = () => { // buttons for workers
        return (
            <div className="jobs-button-container">
                
                <Button onClick={() => {
                        getAndSetAllAvailableJobs();
                        renderHeader('Available Jobs');
                    }}
                    className="jobs-button"
                    color="info"
                    data-testid="available-jobs-menu-item">
                    Available Jobs
                </Button>
                <Button onClick={() => {
                        getAndSetAllAppliedWorkerJobs(currUser.id);
                        renderHeader('My Applications');
                    }}
                    className="jobs-button"
                    color="info" 
                    data-testid="applications-menu-item">
                    Applications
                </Button>
                <Button onClick={() => {
                        getAndSetAllAssignedWorkerJobs(currUser.id);
                        renderHeader('My Assigned Jobs');
                    }} 
                    className="jobs-button"
                    color="info"
                    data-testid="my-jobs-menu-item">
                    My Jobs
                </Button>

            </div>
        );
    };
        
    const renderUserButtons = () => { // buttons for users
        return (
            <div className="jobs-button-container">
               
                <Button onClick={() => {
                        getAndSetActiveUserJobs(currUser.id);
                        renderHeader('My Active Jobs');
                    }} 
                    className="jobs-button"
                    color="info"
                    data-testid="my-jobs-menu-item">
                    Active Jobs
                </Button>
                <Button onClick={() => {
                        getAndSetPendingReviewUserJobs(currUser.id);
                        renderHeader('Jobs Pending Review');
                    }} 
                    className="jobs-button"
                    color="info"
                    data-testid="pending-review-menu-item">
                    Pending Review
                </Button>
                <Button 
                    className="jobs-button" 
                    color="info" 
                    data-testid="jobs-create-job-button" 
                    onClick={toggleCreateJob}
                >
                    Create New Job
                </Button>
            </div>
        );
    };


    // default state of page shows spinner when isLoading is true
    if(isLoading){
        return (
        <div className="spinner">
            <Spinner>
                Loading...
            </Spinner>
        </div>
          )
    }

    return (
        <div className="jobs-container">

            <div className="jobs-header-container">
                {header}
            </div>

            
                <div className="jobs-buttons-container">
                    {buttons}
                </div>
            
                
            
            
                <CreateJob
                    isOpen={showCreateJob} 
                    toggle={toggleCreateJob}
                    onCreate={() => 
                    {
                        setTriggerEffect(!triggerEffect);
                        toggleCreateJob();
                    }}
                    onClose={() => 
                    {
                        toggleCreateJob();
                    }}
                /> 
            
            
            
                <Container>
                    <Row 
                        xs={1} 
                        sm={2} 
                        md={3} 
                        lg={4} 
                        xl={5} 
                        className="justify-content-center"
                    >
                        {jobCards}
                    </Row>
                </Container>
            

            {notFound && (
                <span className="jobs-notFound-message">It looks like no jobs were found.</span>
            )}
        </div>
    )
}

export default Jobs;