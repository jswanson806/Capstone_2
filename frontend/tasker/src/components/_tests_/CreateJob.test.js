import React from 'react';
import {act, render, screen, fireEvent} from '@testing-library/react';
import { UserContext } from '../../helpers/UserContext.js';
import TaskerApi from '../../api.js';
import CreateJob from '../CreateJob.js';
import Jobs from '../Jobs.js';

const userValue = { user: '{"id": 1, "email": "test@email.com", "isWorker": false}' };

const createJob = jest.spyOn(TaskerApi, 'createJob')

const toggleCreateJob = jest.fn();
const onCreate = jest.fn();
const onClose = jest.fn();
const uploadBeforeImage = jest.spyOn(TaskerApi, 'uploadBeforeImage');

describe("createJob form smoke and snapshot tests", () => {

    it("should render without crashing", async () => {

        await act(async () => {render(
            <UserContext.Provider value={userValue}>
                <CreateJob onCreate={onCreate}/>
            </UserContext.Provider>
        )
        });
    });

    it("should match snapshot", async() => {
        let asFragment;

        await act(async () => {const {asFragment: fragment} = render(
            <UserContext.Provider value={userValue}>
                <CreateJob toggleCreateJob={toggleCreateJob}/>
            </UserContext.Provider>
        )
            asFragment = fragment;
        });

        expect(asFragment()).toMatchSnapshot();
    })
});

describe("Handles form input correctly", () => {

    it('updates form input correctly', async () => {
        expect.assertions(3);

        await act(async () => {render(
            <UserContext.Provider value={userValue}>
                <CreateJob onCreate={onCreate}/>
            </UserContext.Provider>
        )
        });

        const titleInput = screen.getByTestId("createJob-form-title-input");
        const bodyInput = screen.getByTestId("createJob-form-body-input");
        const addressInput = screen.getByTestId("createJob-form-address-input")
        fireEvent.change(titleInput, {target: {value: "test title"}});
        fireEvent.change(bodyInput, {target: {value: "test body"}});
        fireEvent.change(addressInput, {target: {value: "test address"}});

        // form input fields should have correct data
        expect(titleInput).toHaveValue("test title");
        expect(bodyInput).toHaveValue("test body");
        expect(addressInput).toHaveValue("test address");

    });

    it('should handle form submission with correct data', async () => {

        expect.assertions(2);

        let jobInfo = {job: {
            title: 'test title',
            body: 'test body',
            address: 'test address',
            posted_by: 1,
            before_image_url: undefined
        }};

        await act(async () => {render(
            <UserContext.Provider value={userValue}>
                <CreateJob onCreate={onCreate}/>
            </UserContext.Provider>
        )
        });

        const titleInput = screen.getByTestId("createJob-form-title-input");
        const bodyInput = screen.getByTestId("createJob-form-body-input");
        const addressInput = screen.getByTestId("createJob-form-address-input")
        fireEvent.change(titleInput, {target: {value: "test title"}});
        fireEvent.change(bodyInput, {target: {value: "test body"}});
        fireEvent.change(addressInput, {target: {value: "test address"}});

        const btn = screen.getByTestId("createJob-submit-button");

        // submit the form
        await act(async () => {
            fireEvent.click(btn);
        });

        // should have called the function with correct data
        expect(createJob).toHaveBeenCalledTimes(1);
        expect(createJob).toHaveBeenCalledWith(jobInfo);


    });

    it('should call api upon form submission', async () => {
        expect.assertions(1);

        await act(async () => {render(
            <UserContext.Provider value={userValue}>
                <CreateJob onCreate={onCreate}/>
            </UserContext.Provider>
        )
        });

        const titleInput = screen.getByTestId("createJob-form-title-input");
        const bodyInput = screen.getByTestId("createJob-form-body-input");
        const addressInput = screen.getByTestId("createJob-form-address-input")
        fireEvent.change(titleInput, {target: {value: "test title"}});
        fireEvent.change(bodyInput, {target: {value: "test body"}});
        fireEvent.change(addressInput, {target: {value: "test address"}});

        const btn = screen.getByTestId("createJob-submit-button");

        // submit the form
        await act(async () => {
            fireEvent.click(btn);
        });

        expect(uploadBeforeImage).toHaveBeenCalledTimes(1);

    });

    it('should call onCreate after submitting', async () => {
        expect.assertions(1);

        await act(async () => {render(
            <UserContext.Provider value={userValue}>
                <CreateJob onCreate={onCreate}/>
            </UserContext.Provider>
        )
        });

        const titleInput = screen.getByTestId("createJob-form-title-input");
        const bodyInput = screen.getByTestId("createJob-form-body-input");
        const addressInput = screen.getByTestId("createJob-form-address-input")
        fireEvent.change(titleInput, {target: {value: "test title"}});
        fireEvent.change(bodyInput, {target: {value: "test body"}});
        fireEvent.change(addressInput, {target: {value: "test address"}});

        const btn = screen.getByTestId("createJob-submit-button");

        // submit the form
        await act(async () => {
            fireEvent.click(btn);
        });

        expect(onCreate).toHaveBeenCalledTimes(1);
    });

});

describe('closing form should call onClose', () => {
    it('works', async () => {
        expect.assertions(1);

        await act(async () => {render(
            <UserContext.Provider value={userValue}>
                <CreateJob onCreate={onCreate} onClose={onClose}/>
            </UserContext.Provider>
        )
        });

        const closeBtn = screen.queryByTestId("createJob-close-button");

        await act(async () => {
            fireEvent.click(closeBtn);
        })

        expect(onClose).toHaveBeenCalledTimes(1);
    })
})