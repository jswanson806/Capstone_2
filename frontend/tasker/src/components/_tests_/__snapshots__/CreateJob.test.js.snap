// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createJob form smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="createJob-container"
    data-testid="createJob-form-container"
  >
    <div>
      <form>
        <label
          for="title"
        >
          Job Title: 
        </label>
        <input
          data-testid="createJob-form-title-input"
          id="title"
          name="title"
          placeholder="Job Title"
          type="text"
          value=""
        />
        <label
          for="body"
        >
          Job Description: 
        </label>
        <textarea
          cols="40"
          data-testid="createJob-form-body-input"
          id="body"
          name="body"
          placeholder="Briefly describe the job..."
          rows="6"
        />
        <label
          for="address"
        >
          Address: 
        </label>
        <input
          data-testid="createJob-form-address-input"
          id="address"
          name="address"
          placeholder="Adress"
          type="text"
          value=""
        />
        <button
          data-testid="createJob-form-button"
          type="submit"
        >
          Post Job
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
